### ISSUE TITLE
[Bug]: @fluentui/react-charting: ERESOLVE unable to resolve dependency tree @testing-library/react

### ISSUE THREAD

**Issue by vgopal99** [microsoft/fluentui#28683]
### Library

React / v8 (@fluentui/react)
[Package: charting](https://github.com/microsoft/fluentui/labels/Package%3A%20charting)

### System Info

```shell
Package: @fluentui/react-charting
```

### Are you reporting Accessibility issue?

no

### Reproduction

30

### Bug Description

We are getting below while running npm install

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: controltower.web@0.1.0
npm ERR! Found: @testing-library/react@14.0.0
npm ERR! node_modules/@testing-library/react
npm ERR!   dev @testing-library/react@"^14.0.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @testing-library/react@"12.1.2" from @fluentui/react-charting@5.16.56
npm ERR! node_modules/@fluentui/react-charting
npm ERR!   @fluentui/react-charting@"^5.16.56" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

Please could you help us to resolve this issue

we don't want to use **--legacy-peer-deps** or **--force**

https://fluentuipr.z22.web.core.windows.net/heads/master/react-charting/demo/index.html

### Logs

```shell
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: controltower.web@0.1.0
npm ERR! Found: @testing-library/react@14.0.0
npm ERR! node_modules/@testing-library/react
npm ERR!   dev @testing-library/react@"^14.0.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @testing-library/react@"12.1.2" from @fluentui/react-charting@5.16.56
npm ERR! node_modules/@fluentui/react-charting
npm ERR!   @fluentui/react-charting@"5.16.56" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
```

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by insaneinside** [comment]
@srmukher, the inclusion of a specific-version peer dependency can easily lead to errors like this, and I'm concerned about the potential ramifications for fixing component governance issues against @testing-library/react when it's specified as such a specific peer dependency.

Can you please expand on your comment "Adding peer dependency for fixing build error" in 9c0267d?  It's not clear to me why a testing tool needs to be a peer dependency, though I'm admittedly unfamiliar with Fluent UI development.  Thank you!

**Comment by AtishayMsft** [comment]
This issue is fixed in version [5.17.0](https://github.com/microsoft/fluentui/tree/@fluentui/react-charting_v5.17.0) of charting library.
